src/plugins/intel_cpu/src/node.cpp

```diff
@@ -548,11 +548,15 @@ void Node::updateShapes() {
                     getTypeStr(),
                     " with name: ",
                     getName());
-    if (needShapeInfer()) {
-        auto result = shapeInfer();
-        if (ShapeInferStatus::success == result.status) {
-            redefineOutputMemory(result.dims);
+    try {
+        if (needShapeInfer()) {
+            auto result = shapeInfer();
```

Comment:
Agree. Error handling limited to redefineOutputMemory stage. It looks better to move try-catch inside methods, but as redefineOutputMemory is overloaded by some nodes, current WA is remaining until future refactoring.

Reply:
The question was about removing the try catch block from `IShapeInfer::Result Node::shapeInfer()` since when the entire `updateShapes` call is inside a try/catch block we don't need to add context to the error messages spawned inside the nested calls.